### ISSUE TITLE
Index out of bounds panic

### ISSUE THREAD

**Issue by Dekker1** [shnarazk/splr#182]
The following fragment of code fails with a panic in the splr solver:

```rust
	#[test]
	fn test_splr() {
		let mut slv = Solver::instantiate(
			&Config::default(),
			&CNFDescription {
				num_of_variables: 4 as usize,
				..CNFDescription::default()
			},
		);
		assert!(slv.add_clause(vec![-1, -2]).is_ok());
		assert!(slv.add_clause(vec![-3, -4]).is_ok());
		assert!(slv.add_assignment(-2).is_ok());
		let a = slv.add_var() as i32;
		assert!(slv.add_clause(vec![1, 3, 4, -a]).is_ok());
		assert!(slv.add_clause(vec![1, -3, -4, -a]).is_ok());
		assert!(slv.add_clause(vec![-1, 3, -4, -a]).is_ok());
		assert!(slv.add_clause(vec![-1, -3, 4, -a]).is_ok());
		assert!(slv.add_clause(vec![-1, -3, -4, a]).is_ok());
		assert!(slv.add_clause(vec![-1, 3, 4, a]).is_ok());
		assert!(slv.add_clause(vec![1, -3, 4, a]).is_ok());
		assert!(slv.add_clause(vec![1, 3, -4, a]).is_ok());
		let b = slv.add_var() as i32;
		assert!(slv.add_clause(vec![1, 3, -b]).is_ok());
		assert!(slv.add_clause(vec![1, 4, -b]).is_ok());
		assert!(slv.add_clause(vec![3, 4, -b]).is_ok());
		assert!(slv.add_clause(vec![-1, -3, b]).is_ok());
		assert!(slv.add_clause(vec![-1, -4, b]).is_ok());
		assert!(slv.add_clause(vec![-3, -4, b]).is_ok());
		assert!(slv.add_clause(vec![-1, -b]).is_ok());
		assert!(slv.add_clause(vec![-a, -b]).is_ok());

		let mut _solns: Vec<Vec<i32>> = slv.iter().collect();
	}
```

It gives the following information:

```
---- tests::test_splr stdout ----
thread 'tests::test_splr' panicked at 'index out of bounds: the len is 5 but the index is 5', /Users/dekker1/.cargo/registry/src/github.com-1ecc6299db9ec823/splr-0.16.0/src/assign/trail_saving.rs:42:17
```

I'm using splr 0.16, and the incremental solver feature using the following declaration in cargo.toml

```
splr = { version = "0.16", features = ["incremental_solver"] }
```

**Comment by shnarazk** [comment]
Oops! Thank again.
0.16.2 contains the fix.

**Comment by Dekker1** [comment]
Great! Thanks again for the quick response and action!

**Comment by shnarazk** [comment]
I've released Splr-0.16.2. I hope 'all' bugs about incremental solver were fixed in this version 😅. Thank you for using my program. 
I'll close this issue and #179 in a few days.
